packages/apollo-link-batch-http/src/batching.ts

```diff
@@ -0,0 +1,118 @@
+import { Observable, Operation, FetchResult } from 'apollo-link-core';
+import { ApolloFetch, GraphQLRequest } from 'apollo-fetch';
+
+import { print } from 'graphql/language/printer';
+
+export interface QueryFetchRequest {
+  operation: Operation;
+
+  // promise is created when the query fetch request is
+  // added to the queue and is resolved once the result is back
+  // from the server.
+  observable?: Observable<FetchResult>;
+  next?: (result: FetchResult) => void;
+  error?: (error: Error) => void;
+  complete?: () => void;
+}
+
+// QueryBatcher doesn't fire requests immediately. Requests that were enqueued within
+// a certain amount of time (configurable through `batchInterval`) will be batched together
+// into one query.
+export default class QueryBatcher {
+  // Queue on which the QueryBatcher will operate on a per-tick basis.
+  public queuedRequests: QueryFetchRequest[] = [];
+
+  private batchInterval: number;
+  private batchMax: number;
+
+  //This function is called to the queries in the queue to the server.
+  private apolloFetch: ApolloFetch;
+
+  constructor({
+    batchInterval,
+    batchMax = 0,
+    apolloFetch,
+  }: {
+    batchInterval: number;
+    batchMax?: number;
+    apolloFetch: ApolloFetch;
```

Comment:
@jbaxleyiii I like that. I think it will be pretty easy to break out the batching into another link. I'll do that with the next commit